File: eco-core/core-plugin/src/main/java/com/willfp/talismans/talismans/util/BlockPlaceListener.java
package com.willfp.talismans.talismans.util;

import org.bukkit.event.EventHandler;
import org.bukkit.event.EventPriority;
import org.bukkit.event.Listener;
import org.bukkit.event.block.BlockPlaceEvent;
import org.jetbrains.annotations.NotNull;

public class BlockPlaceListener implements Listener {
    /**
     * Called on block place.
     * @param event The event to listen for.
     */
    @EventHandler(priority = EventPriority.LOW)
    public void onAttemptTalismanPlace(@NotNull final BlockPlaceEvent event) {
        if (TalismanChecks.getTalismanOnItem(event.getItemInHand()) != null) {
            event.setCancelled(true);
        }
    }
}

File: eco-core/core-plugin/src/main/kotlin/com/willfp/talismans/talismans/util/BlockPlaceListener.kt
package com.willfp.talismans.talismans.util

import org.bukkit.event.EventHandler
import org.bukkit.event.EventPriority
import org.bukkit.event.Listener
import org.bukkit.event.block.BlockPlaceEvent

class BlockPlaceListener : Listener {
    @EventHandler(priority = EventPriority.LOW)
    fun onAttemptTalismanPlace(event: BlockPlaceEvent) {
        if (TalismanChecks.getTalismanOnItem(event.itemInHand) != null) {
            event.isCancelled = true
        }
    }
}